File: crc.py
from zlib import crc32
import sys

def getCRC(file):
    with open(file,'rb') as f:
        return crc32(f.read())& 0xffffffff

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print('You must enter the file')
        exit(1)
    elif len(sys.argv) > 2:
        print('Only one file is permitted')
        exit(1)
    file = sys.argv[1]

    Crc = getCRC(file)
    Dec = int(str(Crc),16)
    print('{:8}{:x}'.format('Hex:',Crc))
    print('{:8}{:x}'.format('Dec:',Dec))

File: crc.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.zip.CRC32;
import java.util.zip.CheckedInputStream;

public class crc {

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java crc <file>");
            System.exit(-1);
        }
        System.out.println(args[0]);
        String path =  args[0];
        String crc = loadCRC32(path);

        System.out.println("HEX:" + crc);
        System.out.println("DEC:"+ Integer.parseInt(crc,16));
    }

    public static String loadCRC32(String filePath) {
        CRC32 crc32 = new CRC32();
        FileInputStream inputStream = null;
        CheckedInputStream checkedinputstream = null;
        String crcStr = null;
        try {
            inputStream = new FileInputStream(new File(filePath));
            checkedinputstream = new CheckedInputStream(inputStream, crc32);
            while (checkedinputstream.read() != -1) {
            }
            crcStr = Long.toHexString(crc32.getValue()).toUpperCase();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (inputStream != null) {
                try {
                    inputStream.close();
                } catch (IOException e2) {
                    e2.printStackTrace();
                }
            }
            if (checkedinputstream != null) {
                try {
                    checkedinputstream.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }

        return crcStr;
    }

}
